sim-lib/src/lib.rs

```diff
@@ -1158,7 +1188,7 @@ async fn produce_simulation_results(
         tokio::select! {
             biased;
             _ = listener.clone() => {
-                return Ok(())
+                break;
```

Comment:
TIL that you can return a value with `break` in rust - perhaps we can do this here? 

`let result = loop ....` and then `break Ok(())` / `break Err(())` and then go on to the joinset wait at the end. Has the advantage of always waiting for everything to close out cleanly vs current approach where we only wait on `Ok()` exits.

Reply:
Yes, that is a great idea! Just added that in the latest commit.